docs/DEVELOPER.md

```diff
@@ -1,12 +1,42 @@
 # Developer Documentation
 
+## Contributing
+Please read the [contributing guidelines here](https://github.com/graycore/daffodil/blob/master/CONTRIBUTING.md).
```

Comment:
Because this is developer documentation, I think it might be better for this link to reference the develop branch version of CONTRIBUTING.md. Just because the master branch hasn't been updated very often. What do you think?

Reply:
@lderrickable To this point, I've set develop as the default branch until we're out of alpha.

Reply:
I saw that you made most links go to the develop branch, but this particular one goes to a 404 page

Reply:
@lderrickable fixed.